### PR TITLE
154 create youglish rest api harness

### PR DESCRIPTION
**Pull request by aedmondson**

### CODE REVIEW

File: ComSemApp/static/ComSemApp/js/youglish.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                    if (status == 400) {
+                        // TODO : Establish standard for these promise resolutions
+                        reject("Bad request");
+                    }
+                    if (status == 404) {
+                        reject("");
+                    }
+                    if (status == 500) {
+                        reject("");
+                    }
```
Comment (dpmacisaac) [inline ComSemApp/static/ComSemApp/js/youglish.js:39]: add else to just reject any non 200

File: ComSemApp/utils.py [hunk 2]
```diff
@@ -70,12 +74,62 @@ def transcribe(request):
...
+        'restricted' : 'yes',
+        'page' : request.GET.get('page', '1'),
+    }
+
+    if not params['query']:
+        return HttpResponseBadRequest()
+    
+    response = get(ENDPOINT, params=params)
+    
+    try:
```
Comment (dpmacisaac) [inline ComSemApp/utils.py:125]: add comment about try

File: ComSemApp/utils.py [hunk 3]
```diff
@@ -70,12 +74,62 @@ def transcribe(request):
...
+                accent (optional) - A code which allows the client to search for a particular accent
+                page   (optional) - Used for pagination to get more results
+        
+        Returns:
+            HttpResponse - A Django HttpResponse object indicating the outcome of the request:
+                JsonResponse           (200) - A success message containing the requested data in JSON format
+                HttpResponseBadRequst  (400) - If the request does not have the required GET arguments
+                Http404                (404) - If the requested phrase has no available video clips
+                HttpResponeServerError (500) - If some problem occurs in communicating with YouGlish
+
```
Comment (dpmacisaac) [inline ComSemApp/utils.py:103]: add into this the format of the json response data

**Review by dpmacisaac — CHANGES_REQUESTED** [review]
bonk

File: ComSemApp/utils.py (hunk 3 above)
Comment (aedmondson) [reply ComSemApp/utils.py:103]: The api is _technically_ subject to change, so I don't want to codify in our codebase and instead link to it

File: ComSemApp/utils.py [hunk 4]
```diff
@@ -70,12 +74,64 @@ def transcribe(request):
             try:
                 text = r.recognize_google(audio)
                 #closes the temp files
-                os.close(in_file_handle)
-                os.close(out_file_handle)
+                close(in_file_handle)
+                close(out_file_handle)
```
Comment (dpmacisaac) [inline ComSemApp/utils.py:78]: why the change?

**Review by dpmacisaac — APPROVED** [review]

File: ComSemApp/utils.py (hunk 4 above)
Comment (aedmondson) [reply ComSemApp/utils.py:78]: close was the only thing being used from os, and I figured this looked a bit cleaner
